src/GUI/GamePanel.java

```diff
@@ -0,0 +1,79 @@
+package GUI;
+
+import Gameplay.*;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.IOException;
+
+public class GamePanel extends JPanel {
+
+    protected static int XOffset = 100;
+    public static Point deckPosition = new Point(620, 20);
+    public static Point tableauPosition = new Point(20, 150);
+    public static int tableauOffset = 100;
+    private static Deck deck;
+    private static WastePile wastePile;
+    private static Foundations[] foundations;
+    private static Tableau[] tableaus;
+    private ShuffleButton shuffleButton;
+
+    public GamePanel() throws IOException {
+        super.setLayout(null);
+        initializePiles();
+
+        GameMoveListener gameMoveListener = new GameMoveListener();
```

Comment:
Regarding above comment, pass `this` to `GameMoveListener` to store the instance of this `GamePanel` in the newly created instance of `GameMoveListener` here.

```suggestion
// GamePanel.java
        GameMoveListener gameMoveListener = new GameMoveListener(this);
```

And replace static references to objects from GamePanel with references from received instance.

```diff
// GameMoveListener.java
public class GameMoveListener extends MouseInputAdapter {
-    private Deck deck;
+    private GamePanel panel;
    // ...
    public void mouseClicked(MouseEvent mouseEvent) {
        // ...
        }else if (pressedComponent instanceof Tableau) {
            // ...
-            for(Foundations foundations: GamePanel.getFoundations()) {
+            for(Foundations foundations: panel.getFoundations()) {
            // ...
    }
```